mlrun/serving/states.py

```diff
@@ -1255,6 +1347,8 @@ def add_step(
             model_endpoint_creation_strategy=model_endpoint_creation_strategy,
             class_args=class_args,
         )
+        if isinstance(class_name, ModelRunnerStep) or class_name == "ModelRunnerStep":
```

Comment:
I think you can move this check to be the first thing you do inside `verify_model_runner_step`